packages/framework-integration-tests/integration/provider-unaware/end-to-end/delete.events.integration.ts

```diff
@@ -0,0 +1,485 @@
+    })
+
+    it('and there are not ReadModels for first product id', async () => {
+      firstProduct = await getProductUndefined(loggedClient, firstProductId)
+      expect(firstProduct).to.be.null
+      firstProductBySkuReadModel = await getProductBySkuUndefined(loggedClient, firstProductSku)
+      expect(firstProductBySkuReadModel).to.be.null
+      firstProductUpdatesReadModel = await getProductUpdatesUndefined(adminTokenClient, firstProductId)
+      expect(firstProductUpdatesReadModel).to.be.null
+      firstSpecialReportsReadModel = await getSpecialReportsReadModelUndefined(specialTokenClient, firstProductId)
+      expect(firstSpecialReportsReadModel).to.be.null
+    })
+
+    it('when find the event for the second product', async () => {
+      events = await queryProductByEntity(loggedClient, secondProductId)
+      productEvent = events.filter((event) => event.entityID === secondProductId)
+      secondProductEvent = productEvent[0]
+    })
+
+    it('then the second event is not changed', async () => {
+      expect(secondProductEvent.deletedAt).to.be.null
+      expect(secondProductEvent.value.productId).to.be.eq(secondProductId)
+    })
+
+    it('and ReadModels for second product id are not deleted', async () => {
+      secondProduct = await getProduct(loggedClient, secondProductId)
+      expect(secondProduct.id).to.be.eq(secondProductId)
+      secondProductBySkuReadModel = await getProductBySku(loggedClient, secondProductSku)
+      expect(secondProductBySkuReadModel.id).to.be.eq(secondProductSku)
+      secondProductUpdatesReadModel = await getProductUpdates(adminTokenClient, secondProductId)
+      expect(secondProductUpdatesReadModel.id).to.be.eq(secondProductId)
+      secondSpecialReportsReadModel = await getSpecialReportsReadModel(specialTokenClient, secondProductId)
+      expect(secondSpecialReportsReadModel.id).to.be.eq(secondProductId)
+    })
+
+    it('and there are not entities for first product but for product 2', async () => {
+      const productsEntities = await findEntities(loggedClient, 'Product')
+      expect(productsEntities.items.some((product) => product.entityID === firstProductId)).to.be.false
+      expect(productsEntities.items.some((product) => product.entityID === secondProductId)).to.be.true
+    })
+
+    it('when find the event for the cart', async () => {
+      events = await queryCartByEntity(loggedClient, firstCartId)
+      const cartEvent = events.filter((event) => event.entityID === firstCartId)
+      firstCartEvent = cartEvent[0]
+    })
+
+    it('then the cart event is not changed', async () => {
+      expect(firstCartEvent.deletedAt).to.be.null
+      expect(firstCartEvent.value.cartId).to.be.eq(firstCartId)
+    })
+
+    it('and ReadModels for the cart id are not deleted', async () => {
+      firstCartReadModel = await getCart(loggedClient, firstCartId)
+      expect(firstCartReadModel.id).to.be.eq(firstCartId)
+    })
```

Comment:
are these `it` tests intended to be run sequentially? As far as I understand, `it`tests inside a `describe` should be independent and able to run in whatever order. I'm not sure if we are executing this here with another mindset

Reply:
Mocha tests run sequentially. In fact, there's [a separate library](https://www.npmjs.com/package/mocha-parallel-tests) that needs to be used to run Mocha tests in parallel.

Reply:
ok